solidity_contracts/erc20/contract.sol

```diff
@@ -1520,10 +1530,11 @@ contract BridgeERC20 is Initializable, ERC20Upgradeable, UUPSUpgradeable {
         // ignore mint and burn
         if (from == address(0) || to == address(0)) return;
         if (_isReservedAddress(to)) {
-            require(msg.value >= egressFee, "Bridge: no enough bridge fee");
-
+            require(msg.value == egressFee, "incorrect egress bridge fee");
             // Call bridgeMessage of EVM Runtime
-            bytes memory n_bytes = abi.encodePacked(to, amount, memo);
+            // using abi.encodePacked: only the last field should be of variable length
```

Comment:
I worry that `abi.encodePacked` is too fragile and instead we should just be doing regular ABI encoding despite the larger size. I am not sure we care that much about the slightly larger size for it to be worth thinking about these limitations in the arguments that can be passed as part of the `bridgeTransfer` message as it may evolve over time.

Reply:
changed from abi.encodePacked to abi.encodeWithSignature